src/msg/consumer/types.go

```diff
@@ -118,6 +118,9 @@ type Options interface {
 // ConsumeFn processes the consumer.
 type ConsumeFn func(c Consumer)
 
+// MessageFn processes the message.
+type MessageFn func(m Message)
```

Comment:
Nit: doc string could maybe use a bit more detail--what sort of processing is it intended to do?

Reply:
It's up to the users how they want to process the message, e.g: it could be indexing the message, or ingest the message.